Yoast/wordpress-seo
# Fix slow custom fields query

## Summary


This PR can be summarized in the following changelog entry:

* Improves the query used to fetch custom fields available as replacement variables.

## Relevant technical choices:

* This uses a simple `SUBSTRING` instead of `LIKE` and `BETWEEN` comparisons. Substring should always use the full index to filter out 100% of matches as opposed to the previous iteration.
* Also removes several unused private functions from replace_vars that features the same query.

The query has the following EXPLAIN:
```
1	SIMPLE	wp_postmeta		index		meta_key	1023		47577992	100	Using where; Using index
```

## Test instructions

This PR can be tested by following these steps:

* Go to the post edit of a post that has custom fields.
* You should still see the same custom fields suggested in the replacement variable editors ( used in the snippet editor ).
* NOTE: The query has also been tested in the related issue by a customer.

## UI changes

## Documentation

## Quality assurance


Fixes #12894
